packages/core/admin/utils/create-cache-dir.js

```diff
@@ -82,26 +83,26 @@ async function createCacheDir({ appDir, plugins, useTypeScript }) {
   // copy admin core code
   await copyAdmin(cacheDir);
 
-  // Copy app.js or app.tsx if typescript is enabled
-  const customAdminConfigJSFilePath = path.join(appDir, 'src', 'admin', 'app.js');
-  const customAdminConfigTSXFilePath = path.join(appDir, 'src', 'admin', 'app.tsx');
-  const customAdminConfigFilePath = useTypeScript
-    ? customAdminConfigTSXFilePath
-    : customAdminConfigJSFilePath;
+  // Retrieve the custom config file extension
+  const customAdminAppConfigFile = await getCustomAppConfigFile(appDir, useTypeScript);
 
-  if (fs.existsSync(customAdminConfigFilePath)) {
+  if (customAdminAppConfigFile !== undefined) {
```

Comment:
I think this could be reduced to, because `.find` either returns a truthy or undefined value. Nothing else. I feel that explicit testing for `undefined` creates the impression there could be another falsy value.

```suggestion
  if (customAdminAppConfigFile) {
```